`py.typed` missing from pypi-published `bazel-runfiles` wheel causes mypy to fail to import

# 🐞 bug report

### Affected Rule

I'm filing this issue with regards to the [bazel-runfiles](https://pypi.org/project/bazel-runfiles) package published to PyPI.

### Is this a regression?

unknown

### Description

I can see [in the wheel](https://pypi.org/project/bazel-runfiles/#bazel_runfiles-1.0.0-py3-none-any.whl) that there is no `py.typed` marker file, even though this file is provided as `data` to the `py_library` [here](https://github.com/bazelbuild/rules_python/blob/main/python/runfiles/BUILD.bazel#L31). This omission causes Mypy to fail to import the package's type hints, and produces an error when running Mypy without `rules_mypy` (such as in my IDE). My understanding of how Mypy works, based on the information [here](https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-library-stubs-or-py-typed-marker) and the fact that `bazel-runfiles` is not a stubs package, is that this `py.typed` file is required in order for Mypy to work. 

Is this working as designed?

## 🔬 Minimal Reproduction

Download the wheel from [here](https://pypi.org/project/bazel-runfiles/#bazel_runfiles-1.0.0-py3-none-any.whl) and observe the absent `py.typed` file. 

## 🔥 Exception or Error

<pre><code>
<!-- If the issue is accompanied by an exception or an error, please share it below: -->
<!-- ✍️-->
ci/bazel/testing/unit/collect_test_results.py:8: error: Skipping analyzing "runfiles": module is installed, but missing library stubs or py.typed marker  [import-untyped]
ci/bazel/testing/unit/collect_test_results.py:8: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
</code></pre>


## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
macOS 15.2
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
Bazelisk version: 1.25.0
INFO: Invocation ID: 6cddecd0-8331-45bd-9d3b-1db39b6d2abb
Build label: 7.4.1
Build target: @@//src/main/java/com/google/devtools/build/lib/bazel:BazelServer
Build time: Mon Nov 11 21:27:36 2024 (1731360456)
Build timestamp: 1731360456
Build timestamp as int: 1731360456
  </code>
</pre>

**Rules_python version:**

<pre>
  <code>
1.0.0
  </code>
</pre>

**Anything else relevant?**

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -67,6 +67,8 @@ END_UNRELEASED_TEMPLATE
 * (pypi) Wheels with BUILD.bazel (or other special Bazel files) no longer
   result in missing files at runtime
   ([#2782](https://github.com/bazel-contrib/rules_python/issues/2782)).
+* (runfiles) The pypi runfiles package now includes `py.typed` to indicate it
+  supports type checking.
 
 {#v0-0-0-added}
 ### Added

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -68,7 +68,8 @@ END_UNRELEASED_TEMPLATE
   result in missing files at runtime
   ([#2782](https://github.com/bazel-contrib/rules_python/issues/2782)).
 * (runfiles) The pypi runfiles package now includes `py.typed` to indicate it
-  supports type checking.
+  supports type checking
+  ([#2503](https://github.com/bazel-contrib/rules_python/issues/2503)).
 
 {#v0-0-0-added}
 ### Added

--- a/python/runfiles/BUILD.bazel
+++ b/python/runfiles/BUILD.bazel
@@ -24,6 +24,7 @@ filegroup(
 
 filegroup(
     name = "py_typed",
+    # See PEP 561: py.typed is a special file that indicates the code supports type checking
     srcs = ["py.typed"],
 )
 

--- a/python/runfiles/BUILD.bazel
+++ b/python/runfiles/BUILD.bazel
@@ -22,13 +22,18 @@ filegroup(
     visibility = ["//python:__pkg__"],
 )
 
+filegroup(
+    name = "py_typed",
+    srcs = ["py.typed"],
+)
+
 py_library(
     name = "runfiles",
     srcs = [
         "__init__.py",
         "runfiles.py",
     ],
-    data = ["py.typed"],
+    data = [":py_typed"],
     imports = [
         # Add the repo root so `import python.runfiles.runfiles` works. This makes it agnostic
         # to the --experimental_python_import_all_repositories setting.
@@ -57,5 +62,8 @@ py_wheel(
     # this can be replaced by building with --stamp --embed_label=1.2.3
     version = "{BUILD_EMBED_LABEL}",
     visibility = ["//visibility:public"],
-    deps = [":runfiles"],
+    deps = [
+        ":py_typed",
+        ":runfiles",
+    ],
 )

